### ISSUE TITLE
Variant call to default member issue with ByRef

### ISSUE THREAD

**Issue by cristianbuse** [WaynePhillipsEA/twinbasic#183]
**Describe the bug**
The default member of a Collection is not recognized if the Collection object is passed ```ByRef As Variant``` to another method.

**To Reproduce**
```VBA
Module Test
    Sub Main()
        Dim c As Collection
        Set c = New Collection
        
        c.Add 1
        c.Add 2
        
        PrintParam c(2) 'Works fine. Prints: 2
        PrintElem c, 2
    End Sub
    
    Private Sub PrintParam(ByVal v As Variant)
        Debug.Print v
    End Sub
    
    Private Sub PrintElem(ByRef vector As Variant, Byval elemIndex As Long)
        PrintParam vector.Item(elemIndex) 'Works fine. Prints: 2
        PrintParam vector(elemIndex)      'Runtime error. Does not recognize default member
    End Sub
End Module
```

**Screenshots**
n/a

**Expected behavior**
The above code works in VBA with no errors.

**Desktop (please complete the following information):**
 - OS: Windows 10 Enterprise V1909
 - twinBASIC compiler v0.10.871

**Additional context**
None

**Comment by WaynePhillipsEA** [comment]
This was a more general problem with the ByRef Variant handling in the runtime call passing to IDispatch

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.10.933, thanks for reporting.
